File: problem-solving/parsers/py/_02.py
def parser(f):
	n = int(f.readline())
	packages = [ [int(i) for i in f.readline().split()] for _ in range(n) ]

	for l in packages:
		assert(len(l) == 2)
	assert(f.readline() == "\n")

	cert = int(f.readline())
	assert(f.readline().strip() == "")

	return cert, (packages,)

def verifier(cert, ans):
	return cert == ans

def error(cert, input, ans):
	print("Input:")
	print("packages = " + str(input[0]))
	print("")
	print("Expected:")
	print(cert)
	print("")
	print("Actual:")
	print(ans)
	print()

File: problem-solving/parsers/java/_02.java
import java.util.Scanner;

public class _02 implements Parser {
    public Object[] parser(Scanner f) {
        int n = f.nextInt();
        int[][] packages = Parser.parse_arrarr_int(f, n, 2);

        assert(f.nextLine() == "");
        assert(f.nextLine() == "");

        int cert = f.nextInt();

        return Parser.ret_parser(cert, (Object)packages);
    }

    public Object run(Object[] input) {
        return Solutions.maxPackages((int[][])input[0]);
    }

    public boolean verifier(Object cert, Object ans) {
        return (int)cert == (int)ans;
    }

    public void error(Object cert, Object[] input, Object ans) {
        System.out.println(String.format(
            "Input:\n" +
            "packages = %s\n" +
            "\n" +
            "Expected:\n" +
            "%d\n" +
            "\n" +
            "Actual:\n" +
            "%d\n",
            Parser.print_arrarr((int[][])input[0]),
            (int)cert,
            (int)ans
        ));
    }

}
